docs/01-app/03-api-reference/03-file-conventions/middleware.mdx

```diff
@@ -12,6 +12,13 @@ The `middleware.js|ts` file is used to write [Middleware](/docs/app/api-referenc
 
 Middleware executes before routes are rendered. It's particularly useful for implementing custom server-side logic like authentication, logging, or handling redirects.
 
+> **Good to know**:
+>
+> - Middleware is built as a separate entry.
+>   - Edge runtime: runs in an isolated environment and does not share `globalThis` or module instances with your server code.
+>   - Node.js runtime: in `next dev` and `next start`, the process may be shared (same `globalThis`). Middleware is compiled as a separate bundle, so module-level singletons aren't shared. Middleware may still be deployed separately by your hosting provider. Avoid relying on in-memory sharing.
```

Comment:
```suggestion
> Middleware is meant to be invoked separately of your render code and in optimized cases deployed to your CDN for fast redirect/rewrite handling, you should not attempt relying on shared modules or globals. 
```
Maybe lets drop the bundling note and just explain it's handled separately.

Reply:
Oh that's a lot better 👍